R/cf.R

```diff
@@ -1024,6 +1024,106 @@ mul.cf <- function(cf, a=1.) {
   return (cf)
 }
 
+#' Exponentiate cf objects
+#'
+#' Note that no complex arithmetic is used, real and imaginary parts are 
+#' treated as seperate and indepenent, such that the real part of cf is
+#' raised to the power of n and similarly for the imaginary part.
+#'
+#' Note that this is generally only allowed on bootstrap samples and mean values,
+#' although it makes sense in some exeptional circumstances. Don't use this
+#' function unless you're certain that you should!
+#' 
+#' @param cf `cf_orig` object.
+#' @param n Numeric.
+#'
+#' @return
+#' The value is
+#' \deqn{cf^n \,.}
+#' @export
+pow.cf <- function(cf, n=1.) {
+  stopifnot(inherits(cf, 'cf_meta'))
+  if(inherits(cf, 'cf_orig')) {
+    cf$cf <- (cf$cf)^n
+    
+    if( has_icf(cf) ){
+      stopifnot(has_icf(cf))
+      cf$icf <- (cf$icf)^n
+    }
+  }
+  
+  if(inherits(cf, 'cf_boot')) {
+    cf$cf.tsboot$t  <- (cf$cf.tsboot$t)^n
+    cf$cf.tsboot$t0 <- (cf$cf.tsboot$t0)^n
+    cf$tsboot.se <- apply(cf$cf.tsboot$t, MARGIN = 2L, FUN = cf$error_fn)
+    cf$cf0 <- cf$cf.tsboot$t0
+  }
+  else cf <- invalidate.samples.cf(cf)
+  return (cf)
+}
+
+#' Assign numeric values n to entries of cf objects
+#'
+#' Note that no complex arithmetic is used, real and imaginary parts are 
+#' treated as seperate and indepenent, such that the real part of cf is
+#' divided by itself and multiplied with n and similarly for the imaginary part.
+#'
+#' Note that this is generally only allowed on bootstrap samples and mean values,
```

Comment:
i don't understand the documentation here... what is this supposed to do?

Reply:
In the FHT ratio function fht_ratio.cf() I calculate the prefactor z/sqrt(z^2 - 1), i.e. pre <- z / pow.cf(z*z - num.cf(z), n=0.5). In this, z is a cf object and consequently z*z is also a cf object, because the objects are multiplied element by element. This means that you cannot simply subtract a numeric value. Therefore, I wrote the function num.cf(), which assigns a numeric value to each entry of the passed cf object, in this case 1, since "num.cf <- function(cf, n=1.)". The result is a cf object with the same dimension as z*z that can be subtracted. So here the value 1 is always subtracted. I achieve the assignment of a numeric value n to all entries via cf$cf <- n*(cf$cf/cf$cf) or cf$icf <- n*(cf$icf/cf$icf). This is what I meant with "the real part of cf is divided by itself and multiplied with n and similarly for the imaginary part.".

Reply:
Now I have rewritten my documentation again and hope that it is now more understandable.

Reply:
Should I remove the check "if( has_icf(cf) )" also in the functions num.cf()? Just like in pow.cf() don't see that it is needed.